nilearn/image/image.py

```diff
@@ -596,3 +599,78 @@ def new_img_like(ref_niimg, data, affine=None, copy_header=False):
         header['cal_max'] = np.max(data) if data.size > 0 else 0.
         header['cal_max'] = np.min(data) if data.size > 0 else 0.
     return ref_niimg.__class__(data, affine, header=header)
+
+
+def threshold_img(img, threshold, mask_img=None,
+                  thresholding_strategy='img_value'):
+    """ Thresholds the given input image based on specific strategy.
+
+    Parameters
+    ----------
+    img: a 3D/4D Niimg-like object
+        Image contains of statistical or atlas maps which should be thresholded.
+
+    threshold: float or string or number
+        If float, we consider it as intensity value and therefore will be used
+        to directly threshold the image meaning thresholding is based on image
+        intensities. The given value should be within the range of minimum and
+        maximum intensity of the image.
+        float option is used with thresholding_strategy='img_value'.
+        If string, it should finish with percent sign e.g. "80%" and
+        should be within the range of "0%" to "100%".
+        If number, we consider it as percentage in real number and should be within
```

Comment:
Again I am not sure which difference you make between float and number.

Also I am a bit confused, it doesn't look like there is anything like the behaviour described by this doc this in the code.

Also thresholding_strategy doesn't seem to be used, or am I missing something here?